src/core/Akka.Remote/RemoteDeploymentWatcher.cs

```diff
@@ -19,8 +19,8 @@ namespace Akka.Remote
     internal class RemoteDeploymentWatcher : ReceiveActor, IRequiresMessageQueue<IUnboundedMessageQueueSemantics>
     {
 
-        private readonly IImmutableDictionary<IActorRef, IInternalActorRef> _supervisors =
-            ImmutableDictionary<IActorRef, IInternalActorRef>.Empty;
+        private readonly IDictionary<IActorRef, IInternalActorRef> _supervisors =
```

Comment:
ImmutableDictionary is what caused the bug (because of how we were using it) with not removing `IActorRef`s for terminated remotely deployed actors. Changed to a good, old-fashioned mutable dictionary instead.